solidity/test/bridge/Bridge.test.ts

```diff
@@ -1423,6 +1448,704 @@ describe("Bridge", () => {
+                        () => {
+                          it("should revert", async () => {
+                            // TODO: Implementation.
+                          })
+                        }
+                      )
+
+                      context(
+                        "when output vector contains a timed out requested redemption with wrong amount",
+                        () => {
+                          it("should revert", async () => {
+                            // TODO: Implementation.
+                          })
+                        }
+                      )
+
+                      context(
+                        "when output vector contains a non-zero P2SH change output",
+                        () => {
+                          // We have this case because P2SH script has a 20-byte
+                          // payload which may match the 20-byte wallet public
+                          // key hash though it should be always rejected as
+                          // non-requested output. There is no need to check for
+                          // P2WSH since the payload is always 32-byte there.
+                          it("should revert", async () => {
+                            // TODO: Implementation.
+                          })
+                        }
+                      )
+
+                      context(
+                        "when output vector contains multiple non-zero change outputs",
+                        () => {
+                          it("should revert", async () => {
+                            // TODO: Implementation.
+                          })
+                        }
+                      )
+
+                      context(
+                        "when output vector contains one change but with zero as value",
+                        () => {
+                          it("should revert", async () => {
+                            // TODO: Implementation.
+                          })
+                        }
+                      )
+
+                      context(
+                        "when output vector contains a non-requested redemption to an arbitrary script hash",
+                        () => {
+                          it("should revert", async () => {
+                            // TODO: Implementation.
+                          })
+                        }
+                      )
+
+                      context(
+                        "when output vector contains a provably unspendable OP_RETURN output",
+                        () => {
+                          it("should revert", async () => {
+                            // TODO: Implementation.
+                          })
+                        }
+                      )
+                    })
+                  }
+                )
+
+                context(
+                  "when wallet state is neither Active nor MovingFunds",
```

Comment:
Just like for the `requestRedemption` case, I'd test them separately.

Reply:
These will be separate sub-contexts within this one. I'll drop a TODO to make it explicit. https://github.com/keep-network/tbtc-v2/pull/128/commits/bd7b48fc10fcb38ee744a3df0676a2919130ab1c